### DISCUSSION TITLE
Live Logging of Jobs does not work behind Apache Reverse Proxy

### DISCUSSION THREAD

**Discussion by Renasb23** [semaphoreui/semaphore#1811] [Q&A] [answered]
Hello,

I have installed and used Ansible Semaphore for a while but have currently changed to it running behind an Apache reverse Proxy for SSL connections.
The entire UI works fine except for the live Logging when you run a job.

Instead of showing the running log and status of the Job it just stays at "waiting" until I reload the Page. Afterwards, I can see the log.

Did I forget to add something?

The Apache Log also throws this error: 
[proxy_http:error] [pid 3335698] (20014)Internal error (specific information not available): [client *.*.*.*:51279] AH01102: error reading status line from remote server localhost:3000
[Wed Mar 06 11:01:37.966326 2024] [proxy:error] [pid 3335698] [client *.*.*.*:51279] AH00898: Error reading from remote server returned by /api/ws

I entered the full Link with which Semaphore is called in the web_host
Apache Config relevant lines:
(Default.conf)
        Redirect permanent /api/ws ws://semaphorehost.de:3000/api/ws
        Redirect permanent /semaphore https://semaphorehost.de

(Default-ssl.conf)
                SSLEngine on
                SSLProxyEngine on
                ProxyPass / http://localhost:3000/
                ProxyPassReverse / http://localhost:3000/

**Comment by fiftin** [comment]
I think problem with Websocket config in Apache

**Comment by Renasb23** [comment, marked as answer]
yes you were correct.
If someone else has the same Issue, this is the necessary Apache Config options:

    <Location />
        ProxyPass http://127.0.0.1:3000/
        ProxyPassReverse http://127.0.0.1:3000/
    </Location>

    <Location /api/ws>
        ProxyPass ws://127.0.0.1:3000/api/ws/
    </Location>

    ProxyPreserveHost off
    ProxyRequests off
    RemoteIPHeader X-Forwarded-For

    <Proxy *>
        Require all granted
    </Proxy>
